ddl/column.go

```diff
@@ -206,9 +225,14 @@ func onAddColumn(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, err error)
 		}
 		// Update the job state when all affairs done.
 		job.SchemaState = model.StateWriteReorganization
+		job.MarkNonRevertible()
 	case model.StateWriteReorganization:
 		// reorganization -> public
 		// Adjust table column offset.
+		offset, err := locateOffsetToMove(columnInfo.Offset, pos, tblInfo)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
```

Comment:
the doc says
> In this situation, the error is guaranteed to be a physical one

but seems this one is not a physical error

Reply:
It is indeed a logical error, but this is not the non-revertible case. When an error is returned, it can be handled by L102-L120: the whole job will be rollbacked.